platform/common/core/generic/vault/rwset.go

```diff
@@ -220,7 +221,7 @@ func entriesEqual[T any](r, o map[string]T, compare func(T, T) bool, nss ...core
 	return nil
 }
 
-func getKeys[V any](m map[core.Namespace]V, nss ...core.Namespace) []string {
+func getKeys[V any](m map[driver.Namespace]V, nss ...driver.Namespace) []string {
 	metaWriteNamespaces := collections.Keys(m)
 	if len(nss) == 0 {
 		return metaWriteNamespaces
```

Comment:
Instead of `collections.Intersection(nss, metaWriteNamespaces)` we could have nss as a Set and define a new method `Set[V].Intersection(s []V) []V`